### DISCUSSION TITLE
注册成 Windows 服务 报错

### DISCUSSION THREAD

**Discussion by MuXia-0326** [1715173329/UnblockNeteaseMusic#385] [Q&A] [answered]
错误信息
E:\UnblockNeteaseMusic_New>node ./nw.js
internal/modules/cjs/loader.js:985
  throw err;
  ^

Error: Cannot find module 'node-windows'
Require stack:
- E:\UnblockNeteaseMusic_New\nw.js
[90m    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:982:15)[39m
[90m    at Function.Module._load (internal/modules/cjs/loader.js:864:27)[39m
[90m    at Module.require (internal/modules/cjs/loader.js:1044:19)[39m
[90m    at require (internal/modules/cjs/helpers.js:77:18)[39m
    at Object.<anonymous> (E:\UnblockNeteaseMusic_New\nw.js:1:17)
[90m    at Module._compile (internal/modules/cjs/loader.js:1158:30)[39m
[90m    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1178:10)[39m
[90m    at Module.load (internal/modules/cjs/loader.js:1002:32)[39m
[90m    at Function.Module._load (internal/modules/cjs/loader.js:901:14)[39m
[90m    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)[39m {
  code: [32m'MODULE_NOT_FOUND'[39m,
  requireStack: [ [32m'E:\\UnblockNeteaseMusic_New\\nw.js'[39m ]
}

**Comment by pan93412** [comment, marked as answer]
記得先安裝 node-windows

```bash
# 先安裝 Node.js
npm i -g yarn
yarn --production
```

**Comment by MuXia-0326** [comment]
现在有一个很奇怪的问题，就是我安装好服务后，用网易云去ping端口也是通的，但网易云就是显示没有网络，不走这个程序的话又是正常的，找不到问题

**Comment by MuXia-0326** [comment]
直接下打包好的程序运行却又可以使用，就很奇怪

**Comment by lemontree1433223** [comment]
还有 npm i node-windows
